### PR TITLE
Add support for BCLinux 8.x and BCLinux-for-Euler 21.10

### PR DESCRIPTION
**Pull request by linhuali**
## Description
- This change added autoinstall support and necessary test updates for BCLinux-8.x and BCLinux-for-Euler 21.10. 
- Tested BCLinux-8.2/8.6 and BCLinux-for-Euler 21.10 with sata/ide/pvscsi/nvme, vmxnet3/e1000e, efi/bios and all passed except known issues.

## Test Log
```
VM information:
+--------------------------------------------------------------------------------+
| Guest OS Distribution     | BigCloud 8.6 x86_64                                |
+--------------------------------------------------------------------------------+
| GUI Installed             | True                                               |
+--------------------------------------------------------------------------------+
| CloudInit Version         | 19.1.17                                            |
+--------------------------------------------------------------------------------+
| Hardware Version          | vmx-20                                             |
+--------------------------------------------------------------------------------+
| Config Guest Id           | other4xLinux64Guest                                |
+--------------------------------------------------------------------------------+
| GuestInfo Guest Id        | other4xLinux64Guest                                |
+--------------------------------------------------------------------------------+
| GuestInfo Guest Full Name | Other 4.x Linux (64-bit)                           |
+--------------------------------------------------------------------------------+
| GuestInfo Guest Family    | linuxGuest                                         |
+--------------------------------------------------------------------------------+
| GuestInfo Detailed Data   | architecture='X86'                                 |
|                           | bitness='64'                                       |
|                           | cpeString='cpe:/o:bclinux:bclinux:8'               |
|                           | distroAddlVersion='8.6 (Core)'                     |
|                           | distroName='BigCloud Enterprise Linux'             |
|                           | distroVersion='8.6'                                |
|                           | familyName='Linux'                                 |
|                           | kernelVersion='4.19.0-372.26.2.el8.bclinux.x86_64' |
|                           | prettyName='BigCloud Enterprise Linux 8.6 (Core)'  |
+--------------------------------------------------------------------------------+

Test Results (Total: 26, Passed: 23, Skipped: 3, Elapsed Time: 03:05:26)
+-------------------------------------------------------------------------+
| ID | Name                                 |   Status        | Exec Time |
+-------------------------------------------------------------------------+
| 01 | deploy_vm_efi_sata_vmxnet3           |   Passed        | 00:28:27  |
| 02 | check_inbox_driver                   |   Passed        | 00:05:38  |
| 03 | ovt_verify_install                   |   Passed        | 00:34:28  |
| 04 | ovt_verify_status                    |   Passed        | 00:06:52  |
| 05 | vgauth_check_service                 |   Passed        | 00:00:46  |
| 06 | check_ip_address                     |   Passed        | 00:00:38  |
| 07 | check_os_fullname                    |   Passed        | 00:00:55  |
| 08 | stat_balloon                         |   Passed        | 00:00:36  |
| 09 | stat_hosttime                        |   Passed        | 00:00:38  |
| 10 | device_list                          |   Passed        | 00:06:31  |
| 11 | check_quiesce_snapshot_custom_script |   Passed        | 00:01:00  |
| 12 | memory_hot_add_basic                 |   Passed        | 00:14:00  |
| 13 | cpu_hot_add_basic                    |   Passed        | 00:09:01  |
| 14 | cpu_multicores_per_socket            |   Passed        | 00:24:31  |
| 15 | check_efi_firmware                   |   Passed        | 00:00:35  |
| 16 | secureboot_enable_disable            | * Not Supported | 00:00:40  |
| 17 | e1000e_network_device_ops            |   Passed        | 00:03:38  |
| 18 | vmxnet3_network_device_ops           |   Passed        | 00:02:48  |
| 19 | pvrdma_network_device_ops            |   Passed        | 00:12:04  |
| 20 | paravirtual_vhba_device_ops          |   Passed        | 00:02:15  |
| 21 | lsilogic_vhba_device_ops             | * Not Supported | 00:00:35  |
| 22 | lsilogicsas_vhba_device_ops          | * Not Supported | 00:00:36  |
| 23 | sata_vhba_device_ops                 |   Passed        | 00:02:36  |
| 24 | nvme_vhba_device_ops                 |   Passed        | 00:02:12  |
| 25 | nvdimm_cold_add_remove               |   Passed        | 00:20:04  |
| 26 | ovt_verify_uninstall                 |   Passed        | 00:01:46  |
+-------------------------------------------------------------------------+
```

### CODE REVIEW

File: autoinstall/BCLinux/BCLinux-for-Euler/21.10/ks.cfg
Comment (keirazhang) [inline autoinstall/BCLinux/BCLinux-for-Euler/21.10/ks.cfg:1]: Please move this file to `autoinstall/BCLinux-for-Euler/21.10/ks.cfg`.

File: autoinstall/README.md [hunk 1]
```diff
@@ -20,6 +20,7 @@
 19. For FreeBSD 13 or later unattend auto-install, please use file FreeBSD/installerconfig.
 20. For Pardus 21.2 Server and XFCE Desktop or later unattend auto-install, please use file Pardus/preseed.cfg.
 21. For openSUSE Leap 15.3 or later unattend auto-install, please use file openSUSE/15/autoinst.xml.
+22. For BCLinux 8.x or BCLinux-for-Euler 21.10 unattend auto-install, please use files under BCLinux.
```
Comment (keirazhang) [inline autoinstall/README.md:23]: Please specify the exact unattend auto-install config file path for BCLinux 8.x and BCLinux-for-Euler 21.10.

File: linux/deploy_vm/collect_vm_logs.yml [hunk 2]
```diff
@@ -13,6 +13,9 @@
       block:
         - name: "Get VM's guest info"
           include_tasks: ../../common/vm_get_guest_info.yml
+          when:
+            - unattend_install_conf is defined
+            - unattend_install_conf | lower is not match('.*euler.*')
```
Comment (keirazhang) [inline linux/deploy_vm/collect_vm_logs.yml:18]: Better to change `euler` to `bclinux-for-euler`.

File: linux/deploy_vm/deploy_vm_from_iso.yml [hunk 3]
```diff
@@ -204,7 +205,7 @@
       include_tasks: ../../common/vm_wait_guest_fullname.yml
       when:
         - unattend_install_conf is defined
-        - unattend_install_conf | lower is not match('.*(minimal|server_without_gui).*')
+        - unattend_install_conf | lower is not match('.*(minimal|server_without_gui|euler).*')
```
Comment (keirazhang) [inline linux/deploy_vm/deploy_vm_from_iso.yml:208]: Better to change `euler` to `bclinux-for-euler`.

**Review by keirazhang — COMMENTED** [review]
Please skip GOSC test cases if GOSC is not supported well by BCLinux. Skip reason should be "Not Supported".

**Comment by keirazhang** [comment]
Since this PR is for supporting BCLinux and BCLinux for Euler. I suggest changing the title to "Add support for BCLinux and BCLinux for Euler". And please don't expose unreleased VMware Tools versions in this PR. Thanks!

**Comment by linhuali** [comment]
> Please skip GOSC test cases if GOSC is not supported well by BCLinux. Skip reason should be "Not Supported".
GOSC tests will be tested by OS Partners manually or semi-automated after applying some workarounds. Thanks.

File: autoinstall/README.md [hunk 4]
```diff
@@ -20,6 +20,8 @@
 19. For FreeBSD 13 or later unattend auto-install, please use file FreeBSD/installerconfig.
 20. For Pardus 21.2 Server and XFCE Desktop or later unattend auto-install, please use file Pardus/preseed.cfg.
 21. For openSUSE Leap 15.3 or later unattend auto-install, please use file openSUSE/15/autoinst.xml.
+22. For BCLinux 8.x unattend auto-install, please use file under BCLinux/8/server_with_GUI/ks.cfg.
+23. For BCLinux-for-Euler 21.10 unattend auto-install, please use file under BCLinux-for-Euler/21.10/ks.cfg.
```
Comment (keirazhang) [inline autoinstall/README.md:24]: Remove 'under' from line 22 and 23. And 'BCLinux/8/server_with_GUI/ks.cfg' can be replaced with 'BCLinux/8/ks.cfg'.

File: autoinstall/BCLinux/8/server_with_GUI/ks.cfg
Comment (keirazhang) [inline autoinstall/BCLinux/8/server_with_GUI/ks.cfg:1]: As there is only 1 ks.cfg for BCLinux, I think it is not necessary to add a directory 'server_with_GUI'. `autoinstall/BCLinux/8/ks.cfg` is enough.

**Review by keirazhang — DISMISSED** [review]
LGTM. Thanks for your contribution!

**Review by keirazhang — APPROVED** [review]

**Pull request merged**